huaweicloud/services/secmaster/resource_huaweicloud_secmaster_playbook_enabling.go

```diff
@@ -0,0 +1,165 @@
+package secmaster
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var nonUpdatableParamsPlaybookEnabling = []string{"workspace_id", "playbook_name", "playbook_id", "active_version_id"}
+
+// @API SecMaster GET /v1/{project_id}/workspaces/{workspace_id}/soc/playbooks/{playbook_id}/versions
+// @API SecMaster PUT /v1/{project_id}/workspaces/{workspace_id}/soc/playbooks/versions/{id}
+func ResourcePlaybookEnabling() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourcePlaybookEnablingCreate,
+		UpdateContext: resourcePlaybookEnablingRead,
+		ReadContext:   resourcePlaybookEnablingUpdate,
+		DeleteContext: resourcePlaybookEnablingDelete,
+
+		CustomizeDiff: config.FlexibleForceNew(nonUpdatableParamsPlaybookEnabling),
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"workspace_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the ID of the workspace to which the playbook version belongs.`,
+			},
+			"playbook_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the playbook ID.`,
+			},
+			"playbook_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the playbook name.`,
+			},
+			"active_version_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: `Specifies the actived playbook version ID.`,
+			},
+			"enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: `Specifies whether the playbook is enabled.`,
+			},
+		},
+	}
+}
+
+func resourcePlaybookEnablingCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	playbookId := d.Get("playbook_id").(string)
+
+	client, err := cfg.NewServiceClient("secmaster", region)
+	if err != nil {
+		return diag.Errorf("error creating SecMaster client: %s", err)
+	}
+
+	bodyParams := map[string]interface{}{
+		"name":              d.Get("playbook_name"),
+		"enabled":           true,
+		"active_version_id": d.Get("active_version_id"),
```

Comment:
It's optional?
```suggestion
		"active_version_id": utils.ValueIgnoreEmpty(d.Get("active_version_id")),
```

Reply:
This field should be a required attribute and has been modified in the schema definition.